webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -249,6 +249,19 @@ func (v validator) validateBootstrap(
 			allErrs = append(allErrs, field.TypeInvalid(p, vAppConfig,
 				"properties and rawProperties are mutually exclusive"))
 		}
+
+		if properties := vAppConfig.Properties; len(properties) != 0 {
```

Comment:
This outer `if` can be removed, and iterate over ` ... := range vAppConfig.Properties {` instead